internal/resource/config/authenticationapi/application/authentication_api_application_resource.go

```diff
@@ -0,0 +1,269 @@
+	authenticationApiApplicationResponse, httpResp, err := r.apiClient.AuthenticationApiAPI.CreateApplicationExecute(apiCreateAuthenticationApiApplication)
+	if err != nil {
+		config.ReportHttpError(ctx, &resp.Diagnostics, "An error occurred while creating an Authentication Api Application", err, httpResp)
+		return
+	}
+
+	// Read the response into the state
+	var state authenticationApiApplicationModel
+	// This is a workaround for the fact that the API does not return the location of oauth client ClientForRedirectlessModeRef
+	if internaltypes.IsDefined(plan.ClientForRedirectlessModeRef) {
+		authenticationApiApplicationResponse, httpResp, err = r.apiClient.AuthenticationApiAPI.GetApplication(config.ProviderBasicAuthContext(ctx, r.providerConfig), plan.ApplicationId.ValueString()).Execute()
+		if err != nil {
+			config.ReportHttpError(ctx, &resp.Diagnostics, "An error occurred while getting an Authentication Api Application", err, httpResp)
+		}
+	}
+
+	diags = readAuthenticationApiApplicationResponse(ctx, authenticationApiApplicationResponse, &state)
+	resp.Diagnostics.Append(diags...)
+
+	diags = resp.State.Set(ctx, state)
+	resp.Diagnostics.Append(diags...)
+}
+
+func (r *authenticationApiApplicationResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var state authenticationApiApplicationModel
+
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	apiReadAuthenticationApiApplication, httpResp, err := r.apiClient.AuthenticationApiAPI.GetApplication(config.ProviderBasicAuthContext(ctx, r.providerConfig), state.ApplicationId.ValueString()).Execute()
+
+	if err != nil {
+		if httpResp != nil && httpResp.StatusCode == 404 {
+			config.ReportHttpErrorAsWarning(ctx, &resp.Diagnostics, "An error occurred while getting an Authentication Api Application", err, httpResp)
+			resp.State.RemoveResource(ctx)
+		} else {
+			config.ReportHttpError(ctx, &resp.Diagnostics, "An error occurred while getting an Authentication Api Application", err, httpResp)
+		}
+	}
+
+	diags = readAuthenticationApiApplicationResponse(ctx, apiReadAuthenticationApiApplication, &state)
+	resp.Diagnostics.Append(diags...)
+
+	// Set refreshed state
+	diags = resp.State.Set(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+}
+
+// Update updates the resource and sets the updated Terraform state on success.
+func (r *authenticationApiApplicationResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+
+	var plan authenticationApiApplicationModel
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	updateAuthenticationApiApplication := r.apiClient.AuthenticationApiAPI.UpdateApplication(config.ProviderBasicAuthContext(ctx, r.providerConfig), plan.Id.ValueString())
```

Comment:
I think we should generally be using the application_id attribute whenever possible, and tread id as just terraform information.